### ISSUE TITLE
Deep Learning for Cosmetics

### ISSUE THREAD

**Issue by sweaterr** [sweaterr/readinglist#375]
https://medium.com/mathematical-beauty/deep-learning-for-cosmetics-2d1b427bbfa2

**Comment by sweaterr** [comment]
At Mira, we build tools that empower beauty enthusiasts to learn, gather inspiration and make informed buying decisions.
Mira에서는 아름다움 애호가가 배우고, 영감을 얻고, 정보를 바탕으로 구매 결정을 내릴 수있는 도구를 만듭니다.

In conversations with over 75 beauty consumers, we’ve learned that one of the foremost challenges that a consumer faces in finding the right products and techniques is identifying authentic and authoritative voices who can speak to their individual concerns.
75 명이 넘는 뷰티 소비자들과의 대화에서 우리는 소비자가 올바른 제품과 기술을 찾는 데 직면하는 가장 중요한 도전 중 하나가 자신의 개인적인 관심사를 말할 수있는 진정성 있고 권위있는 목소리를 알아내는 것을 배웠습니다.

**Comment by sweaterr** [comment]
## The Problem

Let’s formalize the problem: given a set of images of faces, along with a small number of human-labeled images, (eye color, lid shape, etc.) find an intuitive visual similarity metric between eyes (“this beauty guru has eyes similar to yours!”) and a classifier that captures the human-labeled properties.
문제의 형식을 정하자. 적은 수의 인간 레이블이 붙은 이미지 (눈 색깔, 뚜껑 모양 등)와 함께 얼굴 이미지 집합이 눈 사이의 직관적 인 시각적 유사성 메트릭을 찾습니다 ( "이 아름다움 전문가는 눈이 비슷합니다. 너의 것! ")과 사람이 분류 한 특성을 포착하는 분류 자.

In this blog post, we will focus on eye similarity; a follow-up will address classification tasks.
이 블로그 게시물에서는 눈의 유사성에 초점을 맞 춥니 다. 후속 조치는 분류 업무를 다룰 것입니다.

**Comment by sweaterr** [comment]
Raw images are not suited well to either computing visual similarity or performing classifications.
원시 이미지는 시각적 유사성을 계산하거나 분류를 수행하는 데 적합하지 않습니다.

They can contain many superficial similarities (e.g, similar makeup applied, different skin tones washed out by strong lighting, etc.) that are unrelated to eye structure/complexion.
눈 구조 / 피부색과는 무관 한 많은 표면 유사성 (예 : 유사한 메이크업 적용, 강한 조명으로 씻겨내는 다른 피부 색조 등)을 포함 할 수 있습니다.

Furthermore, raw images live in a high dimensional space, requiring a large amount of labeled training data for classification tasks.
또한, 원시 이미지는 높은 차원 공간에 살며 분류 작업을 위해 많은 양의 분류 된 학습 데이터가 필요합니다.

(See the curse of dimensionality).
(차원의 저주 참조).

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39903856-342426dc-5510-11e8-99b6-93dde5c9418e.png)

![image](https://user-images.githubusercontent.com/1518919/39903861-3af1dbc6-5510-11e8-950c-437a4ca3bb76.png)

**Comment by sweaterr** [comment]
Our primary challenge lies in deriving low-dimensional and dense mathematical representations of eye images — known as embeddings — that capture the qualities that we care about and nothing more.
우리의 가장 중요한 도전 과제는 우리가 신경을 써야 할 자질과 그 이상을 포착하는 눈 이미지의 낮은 차원과 밀집된 수학적 표현을 파생시키는 것입니다.

That is, these embeddings should intentionally ignore:.
즉, 이러한 포함은 의도적으로 무시해야합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39903968-bb188886-5510-11e8-998c-05a067ed76a4.png)

**Comment by sweaterr** [comment]
## Image Normalization via Projective Transform

We can eliminate an entire class of superficial similarities with a simple preprocessing step: the projective transform.
우리는 단순한 전처리 단계, 즉 사영 변환을 통해 전체적인 표면 유사성을 제거 할 수 있습니다.

While cropped images of eyes will exhibit many obvious structural differences (e.g, the eye isn’t at the center, or is rotated due to head tilt, etc.), the projective transformation allows us to “warp” images such that the same eye landmarks are guaranteed to occupy the same coordinates.
자른 눈 이미지는 많은 명백한 구조적 차이를 나타낼 것입니다 (예를 들어, 눈이 중심에 있지 않거나 머리 기울임으로 인해 회전하는 경우), 사영 변환은 이미지를 "왜곡 (warp)"하여 동일한 눈 랜드 마크는 동일한 좌표를 차지하도록 보장됩니다.

This is explained well in the scikit-image documentation.
이것은 scikit-image 문서에서 잘 설명되어 있습니다.

With a little bit of linear algebra mathemagic, we can warp an image such that a set of points map to a new, desired shape, rotating and stretching the image in the process:.
선형 대수학 mathemagic을 조금 사용하면 일련의 점을 원하는 모양으로 매핑하고 프로세스에서 이미지를 회전 및 늘이기 같은 이미지를 왜곡 할 수 있습니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39904309-301dd306-5512-11e8-97cb-214dbc774904.png)

**Comment by sweaterr** [comment]
We can apply the same technique to normalizing eye images, rotating/stretching them to a more consistent form.
우리는 똑같은 기술을 눈 이미지를 정규화하고보다 일관된 형태로 회전 / 늘이기에 적용 할 수 있습니다.

We detect facial landmarks using dlib, crop the eyes, and warp them to ensure alignment and consistency.
우리는 dlib를 사용하여 안면 표식을 탐지하고, 눈을 자르고, 정렬하여 일관성과 일관성을 보장합니다.

This preprocessing step significantly improves our ability to converge on embeddings that are invariant to head tilt/pose.
이 전처리 단계는 head tilt / pose에 불변하는 삽입에 수렴하는 능력을 크게 향상시킵니다.

(A detailed overview of this method, when applied to general face alignment, is available here).
(이 방법에 대한 자세한 개요는 일반적인 얼굴 정렬에 적용 할 때 여기에서 사용 가능합니다).

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39904470-f214a372-5512-11e8-8f03-ee8e1f266f49.png)
![image](https://user-images.githubusercontent.com/1518919/39904475-f6650f20-5512-11e8-8825-c664f41234e1.png)

**Comment by sweaterr** [comment]
## Combining Embeddings for Person-level Similarity

With a simple tweak, we can adapt our embeddings to support person-level eye representations as well — abstracting away any noise present in individual frames.
간단한 조작으로 개인의 눈 표현을 지원하기 위해 퍼가기를 적용 할 수 있습니다. 개별 프레임에있는 모든 노이즈를 추상화 할 수 있습니다.

Using our pre-trained weights from the network above, we now adopt a new loss function that rewards the network for placing averages of sets of embeddings from a given individual in close proximity, relative to an unrelated individual, as illustrated below.
위의 네트워크에서 사전 훈련 된 가중치를 사용하여, 아래에 설명 된 것처럼 관련없는 개인과 비교하여 특정 개인의 근접한 집합에 평균 집합을 배치하는 네트워크에 대해 보상하는 새로운 손실 함수를 채택합니다.

**Comment by sweaterr** [comment]
![image](https://user-images.githubusercontent.com/1518919/39905253-522e1d20-5517-11e8-9cbd-ee8165f3e876.png)

**Comment by sweaterr** [comment]
This forms embeddings that can be combined into a more holistic representation of an individual’s eye.
이것은 개인의 눈을보다 전체 론적으로 표현할 수있는 결합을 형성합니다.

While this is a complicated network architecture, it converges quickly due to our usage of transfer learning: appropriating our earlier, related embeddings, trained for single-image similarity.
이것은 복잡한 네트워크 아키텍처이지만 전송 학습의 사용으로 인해 신속하게 수렴됩니다. 즉, 초기 이미지 관련 이미지를 단일 이미지 유사성에 대해 숙지시켜야합니다.

When applied across our dataset of beauty gurus, we find that it produces embeddings that capture fine-grained similarity between individuals, demonstrated below.
미용 전문가의 데이터 세트에 적용하면 개인별로 미세한 유사성을 포착하는 삽입 기능이 생성됩니다.

**Comment by sweaterr** [comment]
# Conclusion and Future Work

Having arrived at high-quality mathematical representations of eyes, in single images and aggregated across multiple views of an individual, we can perform image similarity and retrieval tasks.
고품질의 수학적 표현, 단일 이미지 및 개인의 여러보기에 걸쳐 수집 된 이미지 유사성 및 검색 작업을 수행 할 수 있습니다.

(“Check out these tutorials by people who speak to your concerns”) This is made possible by diligent image preprocessing with the projective transformation, a clever loss function and transfer learning.
( "귀하의 우려를 말한 사람들이이 튜토리얼을 확인하십시오") 이것은 사영 변환, 영리한 손실 기능 및 전송 학습으로 부지런한 이미지 전처리를 통해 가능합니다.

As we will see in a future blog post, these embeddings make several *supervised* machine learning tasks (classifying eye shape, regressing onto eye color, etc.) straightforward and trivial, In addition, when combined with a more naive analysis, they allow us to perform more sophisticated image retrieval tasks, such as searching for similar makeup compositions.
앞으로의 블로그 포스트에서 볼 수 있듯이, 이러한 내장은 여러 가지 * 감독 된 * 기계 학습 작업 (눈 모양 분류, 눈 색깔 퇴각 등)을 직관적이고 사소한 것으로 만듭니다. 또한,보다 단순한 분석과 결합하면 유사한 메이크업 구성을 검색하는 것과 같은보다 정교한 이미지 검색 작업을 수행 할 수 있습니다.

All code and results produced in this post used NumPy, SciPy, Matplotlib, Chainer, dlib, and the SqueezeNet architecture.
이 게시물에서 작성된 모든 코드와 결과에는 NumPy, SciPy, Matplotlib, Chainer, dlib 및 SqueezeNet 아키텍처가 사용되었습니다.

All images displayed above are solely for non-commercial illustrative purposes.
위에 표시된 모든 이미지는 비상업적 인 목적을위한 것입니다.

Feel free to reach out to learn more about our approach and implementation!.
우리의 접근과 구현에 대해 더 자세히 알아 보려면 언제든지 연락하십시오.
